### ISSUE TITLE
How to increase quality?

### ISSUE THREAD

**Issue by platform-kit** [gitmylo/bark-voice-cloning-HuBERT-quantizer#17]
Hey,  @gitmylo, great work on this repo.

If I want to increase the quality what's the best way to go about that?

I imagine the number of steps that are used during both training and inference must be stored in some variable somewhere. Can you point me to it?

Or maybe there's another obvious solution?

**Comment by gitmylo** [comment]
inference steps? i don't know what you're talking about?
this is not an iterative model, it just takes an input, and spits out an output. in one step.

quality is mostly held back by bark's inconsistency, bark can switch speakers on its own, bark can slowly fade to another speaker as well. this can be unpredictable.

because of that, quality greatly depends on how the audio ends, the quality of the audio, the words spoken, and more.

**Comment by platform-kit** [comment]
Oh, I see. I just assumed it was iterative. Thanks for the info.

If I want to train it for TTS purposes to have more consistent quality, with cloned voices of arbitrary sample quality -- is that possible with this model?

Like, would training the core model improve the likelihood of high quality output despite low quality voice cloning samples -- something akin to how 11Labs works?

**Comment by gitmylo** [comment]
no, this model cannot be trained on a specific voice, as what it does doesn't really interact much with the voice.

more info can be found here: https://github.com/gitmylo/audio-webui/wiki/how-bark-works

**Comment by platform-kit** [comment]
Interesting. Are you aware of this HG space?

https://huggingface.co/spaces/kevinwang676/Bark-with-Voice-Cloning

It seems to be be based on this repo, and succeeds at voice cloning.

**Comment by gitmylo** [comment]
I was not aware of it until now, but yes, the code used [in that repo](https://huggingface.co/spaces/kevinwang676/Bark-with-Voice-Cloning/blob/main/cloning/clonevoice.py) is exactly the same as my [example code](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer)

In fact, [these files](https://huggingface.co/spaces/kevinwang676/Bark-with-Voice-Cloning/tree/main/bark/hubert) are still the originals, including the original credit.

The official ways to run are:
* [audio-webui](https://github.com/gitmylo/audio-webui)
  * [audio-webui colab](https://colab.research.google.com/github/gitmylo/audio-webui/blob/master/audio_webui_colab.ipynb)
* [official colab notebook](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/blob/master/colab_notebook.ipynb)
* [regular ipynb notebook](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/blob/master/notebook.ipynb)
* [raw code](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/blob/master/notebook.ipynb)
* [official huggingface space](https://huggingface.co/spaces/GitMylo/bark-voice-cloning)
